index.js

```diff
@@ -11,43 +9,49 @@ export {
   parse,
 };
 
-function parse(html, options) {
+async function parse(html, options) {
   const response = {};
 
   let body = "";
 
-  const esi = new ESI(options)
-    .on("set_response_code", onSetResponseCode)
-    .on("add_header", onAddHeader)
-    .once("set_redirect", onRedirect);
+  const esi = new ESI(options);
+  esi.addEventListener("set_response_code", onSetResponseCode);
+  esi.addEventListener("add_header", onAddHeader);
+  esi.addEventListener("set_redirect", onRedirect);
+
+  const readableStream = ReadableStream.from(html);
+  const htmlTransformer = new HTMLStream({ preserveWS: true });
+  const htmlWriter = new HTMLWriter();
+
+  const transformedStream = readableStream
+    .pipeThrough(htmlTransformer)
+    .pipeThrough(esi)
+    .pipeThrough(htmlWriter);
+
+  const reader = transformedStream.getReader();
+
+  for (;;) {
```

Comment:
```suggestion
  while (true) {
```

kanske en gnutta mer lättläst?

Reply:
Praxis verkar vara att använda `for…of` loopar för att tugga igenom web streams – [docs/api/webstreams.html#async-iteration](https://nodejs.org/dist/latest-v20.x/docs/api/webstreams.html#async-iteration). Varför gör vi inte det?

```js
for await (const chunk of transformedStream) {
  if (chunk.done) break;
  body += chunk.value;
}
```